flow/table.go

```diff
@@ -297,6 +309,22 @@ func (ft *Table) expireAt(now time.Time) {
 	ft.lastExpire = common.UnixMillis(now)
 }
 
+func (ft *Table) unholdAt(now time.Time) {
+	nowMilliseconds := common.UnixMillis(now)
+	for k := range ft.holdedFlows {
+		f, ok := ft.table[k]
+		if ok {
+			if f.FinishType != FlowFinishType_NOT_FINISHED {
+				if nowMilliseconds-f.Last < HoldTimeoutMilliseconds {
+					continue
+				}
+				delete(ft.table, k)
+			}
+		}
+		delete(ft.holdedFlows, k)
+	}
+}
```

Comment:
to have the same behavior you should advise clients like in expire()

if len(expiredFlows) > 0 {
	ft.expireHandler.callback(&FlowArray{Flows: expiredFlows})
}

Reply:
you suggest adding another handler?
I do not see the point in that. Unlike expired flows, flows that were holded have already been reported as finished to the user, so he should not expect to get a second update.